Refactor FXIOS-8407 [v125] Replace linkButton in FakespotActionFooterView

bmihai23 opened the pull request:
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8407)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18634)

## :bulb: Description
- This PR simplifies the implementation of the `FakespotActionFooterView` class by refactoring the link button usage. 
- The changes include replacing the previous implementation with a more straightforward approach using the `LinkButton` class and a `LinkButtonViewModel` for configuration.

**Light Theme**
| Before Dropdown | After Dropdown | 
| ------------- | ------------- |
| ![Simulator Screenshot - iPhone 15 Pro - 2024-02-19 at 11 05 37](https://github.com/mozilla-mobile/firefox-ios/assets/74779930/263bd6d5-72ad-444b-8c92-2c7ddd023fc7) | ![Simulator Screenshot - iPhone 15 Pro - 2024-02-19 at 11 05 40](https://github.com/mozilla-mobile/firefox-ios/assets/74779930/3ad6648f-a34a-4299-8314-4be4d3a8f6b5) |


**Dark Theme**
| Before Dropdown | After Dropdown | 
| ------------- | ------------- |
| ![Simulator Screenshot - iPhone 15 Pro - 2024-02-19 at 11 07 19](https://github.com/mozilla-mobile/firefox-ios/assets/74779930/c1345e1a-7a8c-486e-9341-c35eba539c02) | ![Simulator Screenshot - iPhone 15 Pro - 2024-02-19 at 11 07 21](https://github.com/mozilla-mobile/firefox-ios/assets/74779930/1a339c1f-5c28-40c4-b22a-1709680c3b98) |



## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

thatswinnie commented on firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift:
```diff
@@ -95,9 +95,11 @@ public final class FakespotActionFooterView: UIView, ThemeApplicable {
+    public func applyTheme(theme: Theme) {
+        let colors = theme.colors
+        linkButton.applyTheme(theme: theme)
+        linkButton.setTitleColor(colors.actionPrimary, for: .normal)
```
The title color doesn't have to be set anymore as the `LinkButton` has the correct styles already.

thatswinnie commented on firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift:
```diff
@@ -32,7 +32,7 @@ public struct FakespotActionFooterViewModel {
     private struct UX {
         static let labelSize: CGFloat = 13
-        static let buttonSize: CGFloat = 13
+        static let buttonSize: CGFloat = 16
```
The font size of the button should stay at 13.

thatswinnie reviewed (commented):
Thanks for taking the task on @bmihai23 🙏 

Could you please take a look at the padding of the link? This is the state in the current release which doesn't match the images you posted.
![IMG_0016](https://github.com/mozilla-mobile/firefox-ios/assets/4530/d36976d6-542b-49b0-8368-603943ce987b)

bmihai23 commented:
Hi @thatswinnie, I wanted to reach out and let you know that I've addressed all the modifications you suggested and I've updated the screenshots to reflect the latest changes accurately.

bmihai23 commented on firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift:
```diff
@@ -66,11 +61,17 @@ public final class FakespotActionFooterView: UIView, ThemeApplicable {
+            title: viewModel.actionTitle,
+            a11yIdentifier: viewModel.a11yActionIdentifier,
+            fontSize: UX.buttonSize,
+            contentHorizontalAlignment: .leading
```
Hi @thatswinnie, I've implemented the changes you suggested for the contentInsets in the UX struct. Thank you 🙏

thatswinnie commented:
@bmihai23 could you please update your branch with main? The build is failing because your branch is not up-to-date and there are changes on main that are required by the build system to build correctly.
